docs/reference/src/main/asciidoc/contexts.asciidoc

```diff
@@ -250,3 +250,166 @@ conversation context, as can the current conversation.
+                WeldManager weldManager = CDI.current().select(WeldManager.class).get();
+                BoundRequestContext requestContext = weldManager.instance().select(BoundRequestContext.class, BoundLiteral.INSTANCE).get();
+                BoundSessionContext sessionContext = weldManager.instance().select(BoundSessionContext.class, BoundLiteral.INSTANCE).get();
+                BoundConversationContext conversationContext = weldManager.instance().select(BoundConversationContext.class, BoundLiteral.INSTANCE).get();
+
+                // We will be using bound contexts, prepare backing structures for contexts
+                Map<String, Object> sessionMap = new HashMap<>();
+                Map<String, Object> requestMap = new HashMap<>();
+                BoundRequest boundRequest = new MutableBoundRequest(requestMap, sessionMap);
+
+                // activate contexts
+                requestContext.associate(requestMap);
+                requestContext.activate();
+                sessionContext.associate(sessionMap);
+                sessionContext.activate();
+                conversationContext.associate(boundRequest);
+                conversationContext.activate();
+
+                // propagate all contexts that have some bean in them
+                if (scopeToContextualInstances.get(requestContext.getScope()) != null) {
+                    requestContext.clearAndSet(scopeToContextualInstances.get(requestContext.getScope()));
+                }
+                if (scopeToContextualInstances.get(sessionContext.getScope()) != null) {
+                    sessionContext.clearAndSet(scopeToContextualInstances.get(sessionContext.getScope()));
+                }
+                if (scopeToContextualInstances.get(conversationContext.getScope()) != null) {
+                    conversationContext.clearAndSet(scopeToContextualInstances.get(conversationContext.getScope()));
+                }
+
+                // now execute the actual original task
+                T result = task.call();
+
+                // cleanup, context deactivation, do not trigger @PreDestroy/@Disposes
+                requestContext.deactivate();
+                conversationContext.deactivate();
+                sessionContext.deactivate();
+
+                // all done, return
+                return result;
+            }
+        };
+        return executor.submit(wrappedTask);
+    }
+}
+----------------------------------------------------------------------------------------------------------------------------------------------------------
+
+==== Pitfalls and drawbacks
+
+There are several things that can possibly go wrong when propagating contexts.
+User code needs to be aware that propagation can happen and prepare their beans accordingly.
+For instance request scopes beans could now theoretically be accessed concurrently which hasn't been the case before.
```

Comment:
"request scopes" -> "request scope"
"hasn't been the case before" -> "wasn't the case before" - to stay correct in future versions, not only in the first version including this feature

Reply:
I actually meant `request scoped beans`